storage/storage.go

```diff
@@ -1,9 +1,19 @@
 package storage
 
-import "io"
+import (
+	"io"
+	"os"
+)
+
+type File struct {
```

Comment:
It looks like FileInfo contains the actual Name. See https://golang.org/src/os/types.go?s=391:752#L6 so I don't think you need to make a new struct here. So for like s3 in the list you would create the FileInfo and add the name and add the modified time and then things would just work.